### ISSUE TITLE
Configurations: editing current config multiple times

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#5056]
There is a squish test that checks that the current config can be edited multiple times (by adding a block each time, 30 times). This test is currently failing on the build server because it takes an increasingly long time to save the configuration (it fails once it takes >120 seconds to save a config change).

The gradually increasing times for saving are reproducible on a dev machine.

### Acceptance Criteria

1. Editing same config multiple time does not increase time to save
1. Editing different configs multiple times does not increase time to save
1. Test is reenablled and passes.

**Comment by Tom-Willemsen** [comment]
Once done remove the skip that was added in https://github.com/ISISComputingGroup/System_Tests_UI_E4/pull/32

**Comment by mr-field** [comment]
I was unable to reproduce this on my dev computer.

The relevant parts of the codebase seem to be the `configserver` and `ui.configserver` packages on the GUI side and the block server on the server side. Using the current version of the GUI and block server, the time taken to save and reload a config on my PC is ~10 seconds with no blocks and ~12 seconds with 30 blocks. The save+reload time increases linearly when adding blocks and decreases by the same amount when removing them, so it seems that only the number of blocks is affecting performance, regardless of how many times the config is edited. I tried using older combinations of GUI + block server, but was still unable to reproduce the problem. Saving and reloading configs was slightly slower when using older versions of the block server (~10 seconds with no blocks, ~15 with 30) but the slowdown still wasn't enough to explain this ticket.

I had a look at the code changes to the `configserver` packages and block server between now and the creation date of this ticket, but didn't see much that could have affected performance. The only promising change was a [fix for a memory leak](https://github.com/ISISComputingGroup/ibex_gui/pull/1045) in the config UI, but even when I built the GUI from before that change I could see no slowdown.

I don't have a Squish license so I can't verify if the test passes now; I propose to re-enable it and see what happens. The PR for this is at https://github.com/ISISComputingGroup/System_Tests_UI_E4/pull/42.

**Comment by Tom-Willemsen** [comment]
I ran the test locally and can still reproduce the issue:

![image](https://user-images.githubusercontent.com/24375813/80236512-1458e880-8653-11ea-92a6-1694a241ca95.png)

Let's have a chat about it next week

**Comment by mr-field** [comment]
I am still unable to reproduce this issue outside of the Squish test. I've tried repetedly doing these steps:
- Edit the current config
- Switch to the Blocks tab
- Add a new block for the `DAE:RUNSTATE` PV
- Save the config

However, I did not see a progressive slowdown.

It would be best if this ticket was picked up by someone who can run the Squish tests, so that they can observe what the test is doing and monitor IBEX with a debugger.
